src/libraries/Microsoft.Extensions.Http/src/Logging/LoggingHttpMessageHandler.cs

```diff
@@ -144,11 +147,15 @@ public static void RequestEnd(ILogger logger, HttpResponseMessage response, Time
                 }
             }
 
-            private static string? GetUriString(Uri? requestUri)
+            internal static string? GetUriString(Uri? uri)
             {
-                return requestUri?.IsAbsoluteUri == true
-                    ? requestUri.AbsoluteUri
-                    : requestUri?.ToString();
+                if (uri is null)
+                {
+                    return null;
+                }
+
+                Debug.Assert(uri.IsAbsoluteUri, "HttpClient should not forward relative Uri to its handlers.");
+                return s_logQueryString ? uri.AbsoluteUri : $"{uri.Scheme}://{uri.Host}{uri.AbsolutePath}";
```

Comment:
```suggestion
                if (uri.IsAbsoluteUri)
                {
                    return s_logQueryString
                        ? uri.AbsoluteUri
                        : uri.GetComponents(UriComponents.AbsoluteUri & ~UriComponents.Query, UriFormat.UriEscaped);
                }

                string uriString = uri.ToString();
                int queryOffset = s_logQueryString ? -1 : uriString.IndexOf('?');
                return queryOffset < 0
                    ? uriString
                    : uriString.Substring(0, queryOffset);
```

The handler is public, so it's possible for this to still be a relative Uri https://github.com/dotnet/runtime/pull/63012#discussion_r781199208.

`GetComponents` can save you the intermediate string allocation for the path.